src/HealthCheckResultTask.cs

```diff
@@ -10,3 +10,26 @@ internal static class HealthCheckResultTask
     /// </summary>
     public static readonly Task<HealthCheckResult> Healthy = Task.FromResult(HealthCheckResult.Healthy());
 }
+
+internal class HealthCheckErrorList : List<string>
+{
+    public Task<HealthCheckResult> GetHealthStateAsync(HealthCheckContext context)
+    {
+        if (Count > 0)
+        {
+            return Task.FromResult(new HealthCheckResult(context.Registration.FailureStatus, description: string.Join("; ", this)));
+        }
+
+        return HealthCheckResultTask.Healthy;
+    }
+
```

Comment:
```suggestion
```

Reply:
Also I would switch to extension method for `List<string>` so insteaf of `return errorList?.GetHealthState(context) ?? HealthCheckResult.Healthy();` you can write `return errorList.GetHealthState(context);`